business/namespaces.go

```diff
@@ -38,10 +39,10 @@ func IsAccessibleError(err error) bool {
 	return isAccessibleError
 }
 
-func NewNamespaceService(k8s kubernetes.ClientInterface) NamespaceService {
+func NewNamespaceService(k8s map[string]kubernetes.ClientInterface, kialiSAClients map[string]kubernetes.ClientInterface) NamespaceService {
 	var hasProjects bool
 
-	if k8s != nil && k8s.IsOpenShift() {
+	if k8s != nil && k8s[kubernetes.HomeClusterName].IsOpenShift() {
```

Comment:
I think we want to use the SA here - we should not assume the user will have the permissions to determine if the cluster is openshift. We can assume the SA will have those permissions. (I actually don't remember what IsOpenShift queries, but I thought it looks at some internal resource; in any event, I don't see a reason why we should use the user client here - I think it should be the SA client).

Reply:
This raises an interesting question on whether it is valid to have both openshift and non-openshift clusters in the same mesh? The same is probably true of gateway-api that some clusters may have the CRDs installed and some may not.

Reply:
I was testing in OpenShift and it looks like it is able to get the information. Probably because it uses the Kiali SA when it is required. 
We can change it for kialiSAClients to make it more explicit, anyway.